TypeCobol/Compiler/CodeModel/SymbolTable.cs

```diff
@@ -241,13 +261,33 @@ public List<DataDefinition> GetVariablesByType(DataType dataType, IEnumerable<Da
             return foundedVariables;
         }
 
-        private void SeekVariableType(DataType dataType, DataDefinition variable, ref List<DataDefinition> foundedVariables)
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="dataType"></param>
+        /// <param name="variable"></param>
+        /// <param name="foundedVariables"></param>
+        /// <param name="recursive">No need to recurse by default for variables outside types</param>
+        private void SeekVariableType(DataType dataType, DataDefinition variable, ref ISet<DataDefinition> foundedVariables, bool recursive = false)
```

Comment:
This new param is confusing, it is used only internally by the method to manage its own recursive calls. So the caller should not be aware of it. Anyway as it is a `private` method it's ok to keep it like that.
However I don't understand how this works: for regular data with children don't we need to look in children ? Why is it different for types ?

Reply:
I rewrote the code to be more explicit. I hope it's better.